Scuti/Scripts/UI/Store/Offer Details/PanningAndPinchImage.cs

```diff
@@ -0,0 +1,264 @@
+            else xpos += (int)rectScroll.rect.width / 2;
+
+            if (ypos > 0) ypos = ypos + (int)rectScroll.rect.height / 2;
+            else ypos += (int)rectScroll.rect.height / 2;
+            // Calculate point in scale
+            float widthScroll = rectScroll.rect.width;
+            float heightScroll = rectScroll.rect.height;
+
+            float widthScroll2 = rectImagenToZoom.rect.width;
+            float heightScroll2 = rectImagenToZoom.rect.height;
+
+            // Move image in zoom 
+            rectImagenToZoom.anchoredPosition = new Vector2(widthScroll2 / 2, heightScroll2 / 2) - new Vector2(xpos, ypos);
+            rectImagenToZoom.anchoredPosition = new Vector2(rectImagenToZoom.anchoredPosition.x * (maxZoomDesktop * 0.95f), rectImagenToZoom.anchoredPosition.y * (maxZoomDesktop * 0.95f));
+
+        }
+
+        // ----------------------------------------------------------------------------------------------
+
+        // This method anchors the maximum and minimum values of the scale to mobile
+        private Vector3 ClampDesiredScale(Vector3 desiredScale)
+        {
+            desiredScale = Vector3.Max(initialScale, desiredScale);
+            desiredScale = Vector3.Min(initialScale * maxZoomMobile, desiredScale);
+            return desiredScale;
+        }
+
+        /// <summary>
+        /// IScrollHandler: This method is called when scrolling
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnScroll(PointerEventData eventData)
+        {
+            var delta = Vector3.one * (eventData.scrollDelta.y * zoomSpeed);
+            var desiredScale = transform.localScale + delta;
+
+            desiredScale = ClampDesiredScale(desiredScale);
+
+            transform.localScale = desiredScale;
+        }    
+
+        /// <summary>
+        /// This method will be called on the start of the mouse drag
+        /// </summary>
+        /// <param name="eventData">mouse pointer event data</param>
+        public void OnBeginDrag(PointerEventData eventData)
+        {
+            lastMousePosition = eventData.position;
+            averagePointBetweenTouch = eventData.position;
+        }
+
+        /// <summary>
+        /// This method will be called during the mouse/touch drag
+        /// </summary>
+        /// <param name="eventData">mouse pointer event data</param>
+        public void OnDrag(PointerEventData eventData)
+        {
+            scrollRect.enabled = false;
+
+            Vector2 currentMousePosition = new Vector2(0, 0);
+
+            // Get de position touch
+            if (counterTouch == 1)
+            {
+                currentMousePosition = eventData.position;
+            }
+
+            if (counterTouch >= 2)
+            {
+                currentMousePosition = averagePointBetweenTouch;
+            }
+
+            Vector2 diff = currentMousePosition - lastMousePosition;
+            RectTransform rect = GetComponent<RectTransform>();
```

Comment:
@Jpablogs8 Need to remove this.

At the top of the class make a property like 'public RectTransform Rect'  and then get this property onAwake or Start.   Calling GetComponent shouldn't happen on a method that is called often like OnDrag because it is a very slow method.  So we do it once and keep the reference.